### DISCUSSION TITLE
CFE_ES_WaitForStartupSync() -- behavior if timeout reached?

### DISCUSSION THREAD

**Discussion by CDKnightNASA** [nasa/cFS#895] [Q&A]
I'm a little confused by the CFE_ES_WaitForStartupSync() timeout, there's no way to determine if the timeout was reached (except looking at wall clock time elapsed) and what should the application do if the timeout is reached (if anything)? I'm guessing that the caller should just proceed assuming startup has completed even if the timeout was reached, but perhaps this can be made clear in the documentation for this function?

I plan to use this for SBN but perhaps I should be using CFE_ES_WaitForSystemState() instead? The comments imply CFE_ES_WaitForStartupSync() is a convenience and may be deprecated?

https://github.com/nasa/cFE/blob/1d3e4cdcb67be21032663f1227c8fbe81e9ff5a1/modules/core_api/fsw/inc/cfe_es.h#L460

**Comment by jphickey** [comment]
Yes, `CFE_ES_WaitForStartupSync()` is a simplified API, it does not give any feedback.  The `CFE_ES_WaitForSystemState()` API is the more complete/generic version and it gives the feedback you are looking for (CFE_ES_OPERATION_TIMED_OUT if state was not reached, and CFE_SUCCESS if it was reached).

**Comment by jphickey** [comment]
If the timeout is reached that means some OTHER app is broken, nothing an app can really do about it - just keep going is the typical practice.  (something might be broken, but again, not much of anything that an app can do about it).
